src/views/onboarding/Onboarding.tsx

```diff
@@ -177,6 +178,13 @@ function OnboardingComponent({ productId }: { productId: string }) {
         request_id: requestIdRef.current,
         product_id: productId,
       });
+    } else if ((postLegalsResponse as AxiosError<Problem>).response?.status === 403) {
```

Comment:
move this logic inside the else in order to fill the event considering the 403
then call the setOutcome giving a value based on the event variable

in order to call the setOutcome method just once inside this function, you can move then move setOutcome from row 173 to the if success branch in order to fill it when success